File: VDFEditor/src/test/java/de/treichels/hott/vdfeditor/actions/RenameActionTest.java
package de.treichels.hott.vdfeditor.actions;

import static org.junit.Assert.assertEquals;

import java.util.ArrayList;
import java.util.List;

import org.junit.Before;
import org.junit.Test;

import gde.model.voice.VoiceData;

public class RenameActionTest {
    private List<VoiceData> list;
    private UndoableAction<VoiceData> action;

    @Before
    public void setup() {
        list = new ArrayList<>();
        list.add(new VoiceData("before", null));
    }

    @Test
    public void testRenameAction() {
        action = new RenameAction(0, "after");
        assertEquals(1, list.size());
        assertEquals("before", list.get(0).getName());

        action.apply(list);
        assertEquals(1, list.size());
        assertEquals("after", list.get(0).getName());

        action.undo(list);
        assertEquals(1, list.size());
        assertEquals("before", list.get(0).getName());
    }
}

File: VDFEditor/src/test/kotlin/de/treichels/hott/vdfeditor/actions/RenameActionTest.kt
package de.treichels.hott.vdfeditor.actions

import de.treichels.hott.model.voice.VoiceData
import org.junit.Assert.assertArrayEquals
import org.junit.Test
import kotlin.test.assertEquals

class RenameActionTest {
    @Test
    fun testRenameAction() {
        val list = mutableListOf(VoiceData("before", byteArrayOf(1, 2, 3)))
        val action = RenameAction(0, "after")

        assertEquals(1, list.size.toLong())
        assertEquals("before", list[0].name)
        assertArrayEquals(byteArrayOf(1, 2, 3), list[0].rawData)

        action.apply(list)
        assertEquals(1, list.size.toLong())
        assertEquals("after", list[0].name)
        assertArrayEquals(byteArrayOf(1, 2, 3), list[0].rawData)

        action.undo(list)
        assertEquals(1, list.size.toLong())
        assertEquals("before", list[0].name)
        assertArrayEquals(byteArrayOf(1, 2, 3), list[0].rawData)
    }
}
